### PR TITLE
add missing decal registry options

### LINKED ISSUE #462: add missing decal registry options
### I am planning on implementing this myself

done

### Describe your request

- summit clouds are currently hardcoded to have `scale *= 1.15f`; this should[¹](https://discord.com/channels/403698615446536203/993394511940161546/993395908458852452) be available as a decal registry option `scale` with attributes `multiplyX` and `multiplyY`
- randomised starting frames (used for glitch decals) should be available as `randomiseAnimationOffset`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by microlith57**
adds a scale property (with independent x and y scaling factors), and an animation offset randomisation one (like how glitch decals work)

closes #462

### CODE REVIEW

File: Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs [hunk 1]
```diff
@@ -17,6 +17,11 @@ public static class DecalRegistry {
             { "parallax", delegate(Decal decal, XmlAttributeCollection attrs) {
                 ((patch_Decal)decal).MakeParallax(float.Parse(attrs["amount"].Value));
             }},
+            { "scale", delegate(Decal decal, XmlAttributeCollection attrs) {
+                float scalex = attrs["multiplierX"] != null ? float.Parse(attrs["multiplierX"].Value) : 1f;
```
Comment (microlith57) [inline Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs]: might want to rename to `multiplyX` & `multiplyY`

File: Celeste.Mod.mm/Patches/Decal.cs [hunk 2]
```diff
@@ -116,6 +116,13 @@ private class DecalImage : Component {
             scaredAnimal = true;
         }
 
+        [MonoModIgnore]
+        private float frame;
```
Comment (microlith57) [inline Celeste.Mod.mm/Patches/Decal.cs]: is this the right place for this, ~~and am i using `[MonoModIgnore]` correctly?~~

Comment (bigkahuna443) [reply Celeste.Mod.mm/Patches/Decal.cs]: From a style standpoint, fields generally go at the top of the class. And I'm not sure if having [MonoModIgnore] hurts anything, but don't think it's necessary - if you look at the `scale` variable or the `patch_Player` class, you can generally add private fields into scope w/o issue

File: Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs [hunk 3]
```diff
@@ -130,6 +135,9 @@ public static class DecalRegistry {
                 int[] hiddenFrames = Calc.ReadCSVIntWithTricks(attrs["hiddenFrames"]?.Value ?? "0");
                 ((patch_Decal)decal).MakeScaredAnimation(hideRange, showRange, idleFrames, hiddenFrames, showFrames, hideFrames);
             }},
+            { "randomiseAnimationOffset", delegate(Decal decal, XmlAttributeCollection attrs) {
```
Comment (bigkahuna443) [inline Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs]: `randomizeFrame` might be a better name? The sprite class has a similar option with that name. Also uses proper American/Canadian English 😎 Not sure about the function name, maybe `RandomizeStartingFrame()`?

File: Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs (hunk 1 above)
Comment (bigkahuna443) [reply Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs]: yeah I think those sound better

**Review by bigkahuna443 — APPROVED** [review]

File: Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs (hunk 3 above)
Comment (coloursofnoise) [reply Celeste.Mod.mm/Mod/Registry/DecalRegistry.cs]: > proper American/Canadian English

Don't lump Canadians in there, we use the Queen's English thank you very much (although only when it's convenient 😆)

**Comment by bigkahuna443** [comment]
Latest SC and Sardine7 updates resolve the `scale` conflict, so good to merge.

**Pull request merged**